### DISCUSSION TITLE
Exporting music?

### DISCUSSION THREAD

**Discussion by protbox** [FluidSynth/fluidsynth#1178] [Q&A]
I should preface this with letting you know I'm a musician, and not that great of a developer. It's always been a dream of mine to make a simple to use DAW that utilises soundfonts instead of samples/VSTs, which is why I chose FluidSynth.
So far I've got the UI up and running with a functional piano roll where you can place notes. I've got FluidSynth playing notes from a virtual keyboard and midi controller.

The Question: What I need to do next is figure out how to **get the song from the screen exported into a playable audio format**. If someone could just point me in the right direction (relevant API docs perhaps) that would be greatly appreciated. Sadly a lot of the jargon in the docs doesn't make a lot of sense to me, but if I know what I'm looking for I can read up on it more.

Thanks in advance!

**Comment by jimhen3ry** [comment]
Maybe the VST3 plugin discussed in "bit-exact emulation of FluidSynth in a DAW..." Discussion #1180 would be helpful.

**Reply by protbox** [reply]
Thanks, I will check that out. I did manage to work out my problem using the midi player and file renderer however.
